src/ConnectedMode/Binding/GlobalConfigGenerator.cs

```diff
@@ -0,0 +1,117 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.CSharpVB;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    public interface IGlobalConfigGenerator
+    {
+        string Generate(IEnumerable<SonarQubeRule> rules);
+    }
+
+    internal class GlobalConfigGenerator : IGlobalConfigGenerator
+    {
+        private readonly IEnvironmentSettings environmentSettings;
+
+        private static readonly string inactiveRuleActionText = GetActionText(RuleAction.None);
+
+        public GlobalConfigGenerator() : this(new EnvironmentSettings()) { }
+
+        public GlobalConfigGenerator(IEnvironmentSettings environmentSettings)
+        {
+            this.environmentSettings = environmentSettings;
+        }
+
+        public string Generate(IEnumerable<SonarQubeRule> rules)
+        {
+            if (rules == null)
+            {
+                throw new ArgumentNullException(nameof(rules));
+            }
+
+            var sb = new StringBuilder();
+
+            sb.AppendLine("is_global=true");
+            sb.AppendLine("global_level=500000");
+
+            var sortedRules = rules.OrderBy(r => r.Key);
+
+            foreach (var rule in sortedRules)
+            {
+                var severityText = (rule.IsActive) ? GetActionText(GetVsSeverity(rule.Severity)) : inactiveRuleActionText;
+
+                sb.AppendLine($"dotnet_diagnostic.{rule.Key}.severity = {severityText}");
+            }
+
+            return sb.ToString();
+        }
+
+        internal /* for testing */ RuleAction GetVsSeverity(SonarQubeIssueSeverity sqSeverity)
+        {
+            switch (sqSeverity)
+            {
+                case SonarQubeIssueSeverity.Info:
+                case SonarQubeIssueSeverity.Minor:
+                    return RuleAction.Info;
+
+                case SonarQubeIssueSeverity.Major:
+                case SonarQubeIssueSeverity.Critical:
+                    return RuleAction.Warning;
+
+                case SonarQubeIssueSeverity.Blocker:
+                    return environmentSettings.TreatBlockerSeverityAsError() ? RuleAction.Error : RuleAction.Warning;
+
+                default:
+                    throw new NotSupportedException($"Unsupported SonarQube issue severity: {sqSeverity}");
+            }
+        }
+
+        internal /* for testing */ static string GetActionText(RuleAction ruleAction)
```

Comment:
It wouldn't hurt to include a link to the MS docs page about severities: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-options#severity-level